src/customHooks/useFetch.js

```diff
@@ -0,0 +1,53 @@
+import {useEffect} from "react";
+import axios from "axios";
+
+function useFetch ( dataUrl, setData, setCatchError, setIsLoading ) {
+    // Run the effect on mount
+    useEffect( () => {
+        let isMounted = true;
+
+        // Define request cancellation signal
+        const controller = new AbortController();
+        const { signal } = controller;
```

Comment:
Je had dit samen kunnen voegen
```suggestion
        const { signal, abort } = new AbortController();
```